src/Features/CSharp/Portable/ConvertToRecord/CSharpConvertToRecordCodeFixProvider.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertToRecord
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.ConvertToRecord), Shared]
+    internal class CSharpConvertToRecordCodeFixProvider : CodeFixProvider
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpConvertToRecordCodeFixProvider()
+        {
+        }
+
+        public override CodeAnalysis.CodeFixes.FixAllProvider? GetFixAllProvider()
+        {
+            return WellKnownFixAllProviders.BatchFixer;
```

Comment:
In general the `BatchFixer` is not very performant. It attempts to run all your codefixes and then apply them all at once, failing if any spans overlap. I would have a test the verify that fixall works in general and identify cases where the fixed spans may overlap.

For cases where we need more fine-grained fixall behavior we have the [`SyntaxEditorBasedCodeFixProvider`](https://github.com/dotnet/roslyn/blob/main/src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/SyntaxEditorBasedCodeFixProvider.cs) which allows you to to "simply" write a fix all that is performant while operating on the syntax node level acorss many different iterations of a document.

Reply:
Will address in a follow up PR. Better FixAll support for both the refactoring and code fix is something I'm planning on adding in the future.